### ISSUE TITLE
@NotNull and Optional parsed argument values

### ISSUE THREAD

**Issue by CubBossa** [JorelAli/CommandAPI#440]
### Description

Could we maybe have get(int) in our arguments with @NotNull and as replacement Optional<Object> getOptional(int)? In most cases the argument can't be null and if its supposed to potentially be, one could use the optional getter. An exception could be thrown in the get method if the object is expected to be notnull but is null

### Expected code

Before:
```Java
@Nullable Object get(int index);
```
After:
```Java
@NotNull Object get(int index);
Optional<Object> getOptional(int index);
```

**Comment by DerEchtePilz** [comment]
I would not replace the `@Nullable Object get(int index)` with `@NonNull Object get(int index)` and throw an exception in the `@NonNull` annotated method because we initially implemented it like that and later decided to just let it return `null`.
Also the developer knows it can‘t be `null` if no argument is optional and could just suppress the warning.

However, I really like the idea of the `Optional<Object> getOptional(int index)` because this basically replaces all of the `CommandArguments#getOrDefault` methods because the `Optional` class implements it anyway.

If you or anyone else reading this has any more thoughts I‘d like hearing them!

**Comment by DerEchtePilz** [comment]
@CubBossa I implemented a first implementation of this over [here](https://github.com/JorelAli/CommandAPI/tree/dev/440-getting-arguments)!

**Comment by JorelAli** [comment]
Implemented in 9.0.1.
